Model/Checker/EmailTemplate.php

```diff
@@ -0,0 +1,110 @@
+<?php
+
+namespace SomethingDigital\UpgradeHelper\Model\Checker;
+
+use SomethingDigital\UpgradeHelper\Model\FileIndex;
+use Magento\Email\Model\TemplateFactory as CoreEmailTemplateFactory;
+use Magento\Email\Model\Template\Config;
+use Magento\Framework\App\State as AppState;
+
+class EmailTemplate
+{
+    private $fileIndex;
+
+    private $emailTemplateFactory;
+
+    private $config;
+
+    private $templateMappings = [];
+
+    private $appState;
+
+    public function __construct(
+        CoreEmailTemplateFactory $emailTemplateFactory,
+        Config $config,
+        AppState $appState
+    ) {
+        $this->appState = $appState;
+        $this->emailTemplateFactory = $emailTemplateFactory;
+        $this->config = $config;
+    }
+
+    public function mapEmailTemplates()
+    {
+        $this->appState->setAreaCode(\Magento\Framework\App\Area::AREA_FRONTEND);
```

Comment:
@subbu1993  Just as an FYI @WillC0des  was using the upgrade helper today and go an error "Area code already set". We had to disable this checker for now on one project